### PR TITLE
[extension/sigv4authextension] Make `sts_region` default to `region` if value not provided

### PR DESCRIPTION
**Pull request by erichsueh3**
**Description:** 
This PR makes the value `sts_region` default to the value of `region` if the `sts_region` value is not provided and the `region` value is provided.

**Link to tracking Issue:**
#14573 

**Testing:** <Describe what testing was performed and which tests were added.>
`config_test.go` was updated

**Documentation:** <Describe the documentation added.>
`README.md` was updated

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

**Comment by erichsueh3** [comment]
@mx-psi let me know if any changes are necessary before this can get merged, thanks!

**Review by mx-psi — APPROVED** [review]

File: extension/sigv4authextension/config.go
Comment (mx-psi) [inline extension/sigv4authextension/config.go:46]: Validate should not modify the configuration. Since you are already doing that below, I am not going to block merging this PR because of it, but I would like you to create an issue to move this and the `credsProvider` line below to `Unmarshal`

**Review by mx-psi — APPROVED** [review]

File: extension/sigv4authextension/config.go
Comment (erichsueh3) [reply extension/sigv4authextension/config.go:46]: Will take care of this early next week!

**Pull request merged**
